config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -236,20 +258,56 @@ private void removeExcludedReusablePolicy(Policy policy, AnnotatedBundle annotat
                                 return dependency.getName().equals(PathUtils.extractName(e.getKey()));
                             }
                         }).findFirst();
-                annotatedBundle.getEntities(entityClass).remove(dependency.getName());
-            }
-        }
+        Map entityMap = annotatedBundle.getEntities(entityClass);
+        optionalGatewayEntity.ifPresent(e -> entityMap.put(e.getKey(), e.getValue()));
     }
 
-    private void populateDependentFolders(AnnotatedBundle annotatedBundle, GatewayEntity policyEntity) {
-        if (policyEntity instanceof Folderable) {
-            Folder folder = ((Folderable) policyEntity).getParentFolder();
-            Map<String, Folder> folderMap = annotatedBundle.getEntities(Folder.class);
-            while (folder != null) {
-                folderMap.put(folder.getPath(), folder);
-                folder = folder.getParentFolder();
+    /**
+     * Return TRUE is the Gateway entity needs to be excluded from being loaded.
+     *
+     * @param entityType Type of entity class
+     * @param gatewayEntity Gateway entity to be checked
+     * @param annotatedBundle Annotated Bundle for which bundle is being created.
+     * @param excludeReusable Exclude loading Reusable entities as the dependency
+     * @return TRUE if the Gateway entity needs to be excluded
+     */
+    private boolean excludeGatewayEntity(Class<? extends GatewayEntity> entityType, GatewayEntity gatewayEntity,
+                                         AnnotatedBundle annotatedBundle, boolean excludeReusable) {
+        return excludeReusableEntity(gatewayEntity, excludeReusable)
```

Comment:
Can you please merge excludeReusableEntity and excludePolicy methods into here.

And rewrite the logic like below:
if (<condition1> {
return true;
}

if (<condition2>) {
return true;
}

....

return false;

Reply:
@graju256 Done!